Controller segfaults if `minReplicas` is not set for isvc in RawDeployment mode

/kind bug

**What steps did you take and what happened:**
Build KServe from master at commit `a400bd7f6a3b4d40cc0257e55fb20bb4c8967c9f`.
Deploy using `"defaultDeploymentMode": "RawDeployment"`.

Creat an inference service, e.g. [TensorFlow](https://github.com/kserve/kserve/blob/master/docs/samples/v1beta1/tensorflow/tensorflow.yaml), without setting `minReplicas` in the isvc spec.

The controller segfaults during `createHPA()`:
```shell
{"level":"info","ts":1633356589.2069354,"logger":"inferenceservice-v1beta1-validation-webhook","msg":"validate update","name":"flower-sample"}
{"level":"info","ts":1633356589.206954,"logger":"inferenceservice-v1beta1-validation-webhook","msg":"validate create","name":"flower-sample"}
{"level":"info","ts":1633356589.208686,"logger":"v1beta1Controllers.InferenceService","msg":"Reconciling inference service","apiVersion":"serving.kserve.io/v1beta1","isvc":"flower-sample"}
E1004 14:09:49.310205       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 481 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x19d9560, 0x2a2d4e0)
	k8s.io/apimachinery@v0.19.2/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.19.2/pkg/util/runtime/runtime.go:48 +0x86
panic(0x19d9560, 0x2a2d4e0)
	runtime/panic.go:965 +0x1b9
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa.createHPA(0xc000c580a0, 0x1f, 0x0, 0x0, 0xc000ecc409, 0x7, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go:83 +0x57
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa.NewHPAReconciler(...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go:50
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/autoscaler.createAutoscaler(0x1edacc8, 0xc0005d1680, 0xc000499ce0, 0xc000c580a0, 0x1f, 0x0, 0x0, 0xc000ecc409, 0x7, 0x0, ...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/autoscaler/autoscaler_reconciler.go:79 +0x208
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/autoscaler.NewAutoscalerReconciler(0x1edacc8, 0xc0005d1680, 0xc000499ce0, 0xc000c580a0, 0x1f, 0x0, 0x0, 0xc000ecc409, 0x7, 0x0, ...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/autoscaler/autoscaler_reconciler.go:50 +0xa8
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/raw.NewRawKubeReconciler(0x1edacc8, 0xc0005d1680, 0xc000499ce0, 0xc000c580a0, 0x1f, 0x0, 0x0, 0xc000ecc409, 0x7, 0x0, ...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/raw/raw_kube_reconciler.go:44 +0xa8
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/components.(*Predictor).Reconcile(0xc0005da500, 0xc0006c0800, 0x13, 0x1ecf990)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/components/predictor.go:109 +0xa98
github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice.(*InferenceServiceReconciler).Reconcile(0xc000d09b40, 0x1ec5558, 0xc000c3a2d0, 0xc00095ef09, 0x7, 0xc00095eef0, 0xd, 0xc000c3a2d0, 0xc0007de000, 0x1a824c0, ...)
	github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/controller.go:141 +0x4c6
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000b6a780, 0x1ec54b0, 0xc0000a3000, 0x1a3ae60, 0xc000611ac0)
	sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:263 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000b6a780, 0x1ec54b0, 0xc0000a3000, 0x200c0007b0100)
	sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:235 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1(0x1ec54b0, 0xc0000a3000)
	sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:198 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000939750)
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0011c5f50, 0x1e8d460, 0xc0000ad1d0, 0xc0000a3001, 0xc0005b5560)
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000939750, 0x3b9aca00, 0x0, 0xc00001a401, 0xc0005b5560)
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1ec54b0, 0xc0000a3000, 0xc000e9c360, 0x3b9aca00, 0x0, 0xc0004fc101)
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x1ec54b0, 0xc0000a3000, 0xc000e9c360, 0x3b9aca00)
	k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:195 +0x497
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1710997]
```


**What did you expect to happen:**
Controller can successfully created the HPA without crashing.

**Anything else you would like to add:**
```Go
type ComponentExtensionSpec struct {
	// Minimum number of replicas, defaults to 1 but can be set to 0 to enable scale-to-zero.
        // +optional
	MinReplicas *int `json:"minReplicas,omitempty"`
```

Since `minReplicas` is optional and a pointer, no value is assigned if left empty by a user, resulting in the crash.
This can easily be fixed by checking if `componentExt.MinReplicas` is `nil` :
```Go
func createHPA(componentMeta metav1.ObjectMeta,
	componentExt *v1beta1.ComponentExtensionSpec) *v2beta2.HorizontalPodAutoscaler {
	var minReplicas int32
	if componentExt.MinReplicas == nil || int32(*componentExt.MinReplicas) < int32(constants.DefaultMinReplicas) {
		minReplicas = int32(constants.DefaultMinReplicas)
	} else {
		minReplicas = int32(*componentExt.MinReplicas)
	}
        //...
```

**Environment:**

- Istio Version: /
- Knative Version: /
- KFServing Version: commit `a400bd7f6a3b4d40cc0257e55fb20bb4c8967c9f`
- Kubeflow version: /
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
```shell
minikube version: v1.18.1
commit: 09ee84d530de4a92f00f1c5dbc34cead092b95bc
```
- Kubernetes version: 
```shell
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04.6 LTS

--- a/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
+++ b/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
@@ -80,7 +80,6 @@ func getHPAMetrics(metadata metav1.ObjectMeta) []v2beta2.MetricSpec {
 
 func createHPA(componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec) *v2beta2.HorizontalPodAutoscaler {
-	var minReplicas int32 
 	if componentExt.MinReplicas == nil || (*componentExt.MinReplicas) < constants.DefaultMinReplicas {
 		minReplicas = int32(constants.DefaultMinReplicas)
 	} else {

--- a/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
+++ b/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
@@ -80,10 +80,13 @@ func getHPAMetrics(metadata metav1.ObjectMeta) []v2beta2.MetricSpec {
 
 func createHPA(componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec) *v2beta2.HorizontalPodAutoscaler {
-	minReplicas := int32(*componentExt.MinReplicas)
-	if minReplicas < int32(constants.DefaultMinReplicas) {
+	var minReplicas int32 
+	if(componentExt.MinReplicas == nil || (*componentExt.MinReplicas) < constants.DefaultMinReplicas) {
 		minReplicas = int32(constants.DefaultMinReplicas)
+	} else {
+		minReplicas = int32(*componentExt.MinReplicas)
 	}
+	
 	maxReplicas := int32(componentExt.MaxReplicas)
 	if maxReplicas < minReplicas {
 		maxReplicas = minReplicas

--- a/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
+++ b/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
@@ -81,7 +81,7 @@ func getHPAMetrics(metadata metav1.ObjectMeta) []v2beta2.MetricSpec {
 func createHPA(componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec) *v2beta2.HorizontalPodAutoscaler {
 	var minReplicas int32 
-	if(componentExt.MinReplicas == nil || (*componentExt.MinReplicas) < constants.DefaultMinReplicas) {
+	if componentExt.MinReplicas == nil || (*componentExt.MinReplicas) < constants.DefaultMinReplicas {
 		minReplicas = int32(constants.DefaultMinReplicas)
 	} else {
 		minReplicas = int32(*componentExt.MinReplicas)

--- a/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
+++ b/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go
@@ -80,6 +80,7 @@ func getHPAMetrics(metadata metav1.ObjectMeta) []v2beta2.MetricSpec {
 
 func createHPA(componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec) *v2beta2.HorizontalPodAutoscaler {
+	var minReplicas int32
 	if componentExt.MinReplicas == nil || (*componentExt.MinReplicas) < constants.DefaultMinReplicas {
 		minReplicas = int32(constants.DefaultMinReplicas)
 	} else {

